bip-tap-vm.mediawiki

```diff
@@ -73,13 +73,18 @@ splits are validated.
 Input mapping is only executed for state transitions that specify
 <code>prev_asset_witnesses</code>.
 
-Given a set of inputs, each identified by a <code>prev_asset_id</code>, the
+Given a set of inputs, each identified by a <code>prev_asset_input</code>, the
 input commitment (which is used as the previous output) is constructed as
 follows:
 
 # Initialize a new empty MS-SMT tree as specified in [[./bip-tap-ms-smt.mediawiki|bip-tap-ms-smt]].
 # For each Taproot Asset input ''c_i'', identified in the <code>prev_asset_witnesses</code> field:
-## Serialize the referenced previous asset leaf (identified by <code>prev_outpoint || asset_id || asset_script_hash</code>) in TLV format.
+## Serialize the referenced previous asset leaf (identified by <code>prev_outpoint || prev_asset_id || prev_asset_script_hash</code>) in TLV format.
+## From asset version 1:
+### For a minting transaction, a copy of the output leaf with emptied <code>prev_asset_witnesses</code> is used, in addition to these modifications:
+#### If the minted asset has a group key, the <code>asset_script_hash</code> of the copied leaf should be set equal to the group key. This enforces that the state transition verification uses the group key when validating the spend.
+#### If the asset has no group key, the <code>asset_script_hash</code> field should be blank. This will short-circuit the state transition verification, allowing minting an asset that does not support emission.
```

Comment:
I don't think this modification of `asset_script_hash` is actually necessary?

Right now we short circuit based on `HasGenesisWitness()`, with no change for grouped assets. I think for grouped asset minting we would want to NOT short circuit if the asset passes `HasGenesisWitness()` and also has a group key.

With that change, I don't think we can just swap in the group key later on in the VM vs. modifying the `asset_script_hash`.

Reply:
Unconditional short circuiting here:

https://github.com/lightninglabs/taproot-assets/blob/257ad39d53a04f1cf05d6c4abf30d972ede67da4/vm/vm.go#L286

Reply:
As discussed offline, would be good to be more explicit in describing that the actual asset leaf is not modified here, only the _copy_ made when performing actual virtual transaction validation.

It would also be good to spell out `is_valid_issuance_txn()` explicitly.

Reply:
Yeah I don't think we need to do anything different here. If this is a minting event w/o a group key, then we don't need to do anything.